server/src/main/java/io/crate/analyze/relations/FullQualifiedNameFieldProvider.java

```diff
@@ -47,13 +48,29 @@
     private final Map<RelationName, AnalyzedRelation> sources;
     private final ParentRelations parents;
     private final String defaultSchema;
+    private final boolean errorOnUnknownObjectKey;
 
     public FullQualifiedNameFieldProvider(Map<RelationName, AnalyzedRelation> sources,
                                           ParentRelations parents,
-                                          String defaultSchema) {
+                                          String defaultSchema,
+                                          boolean errorOnUnknownObjectKey) {
         this.sources = Objects.requireNonNull(sources, "Please provide a source map.");
         this.parents = Objects.requireNonNull(parents, "ParentRelations must not be null");
         this.defaultSchema = Objects.requireNonNull(defaultSchema, "Default schema must not be null");
+        this.errorOnUnknownObjectKey = errorOnUnknownObjectKey;
+    }
+
+    private Symbol getField(AnalyzedRelation sourceRelation, ColumnIdent columnIdent, Operation operation, boolean errorOnUnknownObjectKey) {
+        Symbol newField;
+        try {
+            newField = sourceRelation.getField(columnIdent, operation, errorOnUnknownObjectKey);
+        } catch (ColumnUnknownException | AmbiguousColumnException e) {
+            throw e;
+        } catch (RuntimeException e) {
```

Comment:
Would it be possible to narrow the types of exceptions where we fallback to `NULL` ?

Reply:
I assume you mean to not use `RuntimeException` aligned with the next comment